src/utils/useFetch.ts

```diff
@@ -86,12 +86,14 @@ const useFetch = <T = unknown>(
     };
 
     fetchData();
+  }, [url]);
 
+  useEffect(() => {
     // runs when component is destroyed.
     return () => {
       cancelRequest.current = true;
     };
-  }, [url]);
+  }, []);
```

Comment:
i wonder why there was a URL here in the first place, there shouldn't be one

Reply:
As we want this hook to be called only when there is a change in the URL to make a request again.